model-datastructure/src/commonMain/kotlin/org/modelix/model/LinearHistory.kt

```diff
@@ -5,7 +5,14 @@ import org.modelix.model.lazy.IDeserializingKeyValueStore
 import org.modelix.model.lazy.KVEntryReference
 import org.modelix.model.persistent.CPVersion
 
-class LinearHistory(val baseVersionHash: String?) {
+/**
+ * Was introduced in https://github.com/modelix/modelix/commit/19c74bed5921028af3ac3ee9d997fc1c4203ad44
+ * together with the UndoOp. The idea is that an undo should only revert changes if there is no other change that relies
+ * on it. In that case the undo should do nothing to not indirectly undo newer changes.
+ * For example, if you added a node and someone else started changing properties on the that node, your undo should not
+ * remove the node to not lose the property changes.
+ */
+class SlowLinearHistory(val baseVersionHash: String?) {
```

Comment:
I don't understand the naming of the class given the comment. Is this a special implementation that support undo operations and is only accidentally slow due to the algorithm?